### PR TITLE
Dropbox integration [#4]

### PR DESCRIPTION
**Pull request by jwright**
Connect to Dropbox instead of using the local file system.

Closes issue #4
- Updated README with how to setup Dropbox
- Created a support file to stub out Dropbox API calls
- Added Rake task to be able to retrieve a user's Dropbox access token
- Dropbox reader to read files from a Dropbox store
- Dropbox iterator to iterate over files from a Dropbox store
- Dropbox output store to save files in a Dropbox store

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
+gem 'launchy'
 gem 'coveralls', require: false
 
 gem 'redcarpet'
 gem 'nokogiri'
+gem 'dropbox-sdk'
 
 group :development, :test do
   gem 'rspec'
+  gem 'webmock', '~> 1.13.0'
```
Comment (keiththomps) [inline Gemfile]: Is this a dependency of fake web? If not we should remove it.

File: lib/dropdown/processor.rb [hunk 2]
```diff
@@ -1,22 +1,24 @@
...
 
-    def markdown_iterator=(value)
-      @iterator = value
-    end
-
-    def markdown_iterator
-      @iterator ||= Iterators::FileIterator.new(source)
+    def storage=(value)
+      @storage = value
+      self.reader = Dropdown::Readers::ReaderFactory.create(value)
```
Comment (keiththomps) [inline lib/dropdown/processor.rb:19]: Should we be consistent in how we are accessing these attributes?

Comment (jwright) [inline lib/dropdown/processor.rb:19]: Good catch. Removed.

File: spec/dropdown/iterators/dropbox_iterator_spec.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  end
+
+  describe '.each' do
+    include DummyDropbox
+
+    let(:access_token) { 'blah' }
+    let(:dir) { 'Documents/test' }
+    before { Dropdown.configure { |c| c.dropbox_access_token = access_token } }
+
+    it 'should iterate over markdown files that end in .md' do
```
Comment (keiththomps) [inline spec/dropdown/iterators/dropbox_iterator_spec.rb]: Here's a super nit-pick (that I don't think needs changed, just funny). This appears to be the only file that uses "it should" for tests

Comment (jwright) [inline spec/dropdown/iterators/dropbox_iterator_spec.rb]: Changed!

**Pull request merged**